api/application.go

```diff
@@ -1108,6 +1101,28 @@ func (h ApplicationHandler) AssessmentCreate(ctx *gin.Context) {
 	h.Respond(ctx, http.StatusCreated, r)
 }
 
+// tagMap returns a map of applicationTags indexed by application id.
+func (h *ApplicationHandler) tagMap(
+	ctx *gin.Context,
+	applications []model.Application) (mp map[uint][]model.ApplicationTag, err error) {
+	ids := []uint{}
+	for i := range applications {
+		app := &applications[i]
+		ids = append(ids, app.ID)
+	}
+	mp = make(map[uint][]model.ApplicationTag)
+	list := []model.ApplicationTag{}
+	db := h.DB(ctx)
+	err = db.Find(&list, "ApplicationID", ids).Error
+	if err != nil {
+		return
+	}
+	for _, tag := range list {
+		mp[tag.ApplicationID] = append(mp[tag.ApplicationID], tag)
+	}
+	return
+}
+
```

Comment:
Re: https://github.com/konveyor/tackle2-hub/blob/main/api/application.go#L175  This omits the:
```
h.preLoad(h.DB(ctx), clause.Associations)
```
I don't see why we'd need the associations.  In fact, it seem like it would load a ton of unwanted data.

Reply:
The tag needs to be fetched for the ref.name.
This can be optimized though to prevent fetching the application.